### PR TITLE
build and integrate a cacher for full self objects

### PR DESCRIPTION
**Pull request by maxtaco**
- useful for the many one-off access of the full self object on startup

### CODE REVIEW

**Comment by maxtaco** [comment]
cc: @mmaxim 

This PR should eliminate most if not all of unnecessary user loads.  It will make things faster for people like @malgorithms who were previously spending lots of time reloading their own user object over and over.

File: go/engine/list_trackers.go [hunk 1]
```diff
@@ -102,10 +102,12 @@ func (e *ListTrackersEngine) ensureUID() error {
...
-	user, err := libkb.LoadUser(libkb.NewLoadUserByNameArg(e.G(), e.username))
-	if err != nil {
-		return err
-	}
-	e.uid = user.GetUID()
-	return nil
+
+	arg := libkb.NewLoadUserByNameArg(e.G(), e.username)
+	var err error
+	err = e.G().FullSelfCacher.WithUser(arg, func(user *libkb.User) error {
```
Comment (patrickxb) [inline go/engine/list_trackers.go]: This isn't always self.  `keybase list-trackers someoneelse` will run this.

File: go/engine/list_tracking.go [hunk 2]
```diff
@@ -59,33 +59,34 @@ func (e *ListTrackingEngine) RequiredUIs() []libkb.UIKind { return []libkb.UIKin
...
-		user, err = libkb.LoadMe(libkb.NewLoadUserArg(e.G()))
-	}
-	if err != nil {
-		return
+		arg.Self = true
 	}
 
-	var trackList TrackList
-	trackList = user.IDTable().GetTrackList()
+	err = e.G().FullSelfCacher.WithUser(arg, func(user *libkb.User) error {
```
Comment (patrickxb) [inline go/engine/list_tracking.go]: This isn't always self either.  `keybase list-tracking someoneelse`

Comment (patrickxb) [inline go/engine/list_tracking.go]: I think I get it....this won't cache if not self, and won't use self if not self.

File: go/engine/list_trackers.go (hunk 1 above)
Comment (patrickxb) [inline go/engine/list_trackers.go]: Oh, I see.

**Comment by maxtaco** [comment]
Yeah I wanted to allow the same program structure if self or otherwise. It is confusing and needs more comments. Will add. Also the kbfs failure is a problem that I need to look into. Seems to be deadlock along some path.

**Comment by patrickxb** [comment]
Just was thinking: the isSelfLoad check will always fail to match me if arg.Username is a social assertion.

**Comment by maxtaco** [comment]
Agreed, but I think that's OK, right?  It just would mean a cache miss.

**Comment by patrickxb** [comment]
Yup.

**Comment by maxtaco** [comment]
Approach will be to try to duplicate the deadlock outside of kbfs-docker tests. Try [this](https://ci.keyba.se/job/kbfs-server/job/master/4141/execution/node/181/log/?consoleFull)

**Comment by patrickxb** [comment]
woohoo!

**Comment by maxtaco** [comment]
haha, will, it passed C/I twice in a row!

**Pull request merged**